sparse/_utils.py

```diff
@@ -168,19 +290,31 @@ def random(
     if data_rvs is None:
         data_rvs = random_state.rand
 
-    # Use the algorithm from python's random.sample for k < mn/3.
-    if elements < 3 * nnz:
-        ind = random_state.choice(elements, size=nnz, replace=False)
+    if nnz == elements or density >= 1:
+        ind = np.arange(elements)
+    elif nnz < 2:
+        ind = random_state.choice(elements, nnz)
+    # Faster to find non-sampled indices and remove them for dens > .5
+    elif elements - nnz < 2:
+        ind = reverse(random_state.choice(elements, elements - nnz), elements)
+    elif nnz > elements / 2:
+        nnztemp = elements - nnz
+        # Using algorithm A for dens > .1
+        if elements > 10 * nnztemp:
+            ind = reverse(
+                algD(nnztemp, elements, random_state.choice(np.iinfo(np.int32).max)),
```

Comment:
Have you tried changing these to `int64`?

Reply:
That was my original implementation before commit  #2bdd17a, but np.choice can only take 32 bit integers for the seed